src/Symfony/Component/DependencyInjection/ContainerInterface.php

```diff
@@ -30,6 +30,8 @@
     /**
      * Sets a service.
      *
+     * It should be considered harmful to replace an existing service with a new instance, as it can break the dependency graph.
```

Comment:
Maybe it should be rephrased a bit. Technically the current behavior is as follow;

```php
$container->set('kernel', $kernel); // allowed; not initialized yes
$container->set('kernel', $newKernel); // disallowed; already initialized
```

Eventually we allow overwriting anything before initialization. Technically this implementation (hence comment in `ContainerInterface`) is optional, but we do implement it in `Container` by design.